src-self-hosted/libc_installation.zig

```diff
@@ -34,6 +38,52 @@ pub const LibCInstallation = struct {
         ZigIsTheCCompiler,
     };
 
+    fn envSubstitution(allocator: *Allocator, input: []const u8, stderr: anytype) ![:0]u8 {
+        var result = try ArrayList(u8).initCapacity(allocator, input.len + 1);
+        errdefer result.deinit();
+
+        var i: usize = 0;
+        outer: while (i < input.len) : (i += 1) {
+            if (input[i] == '$') {
+                for (input[i + 1..]) |c, j| {
+                    const input_offset = i + j + 1;
+                    if ((!ascii.isAlNum(c) and c != '_') or input_offset == input.len - 1) {
+                        const needle = if (input_offset == input.len - 1)
```

Comment:
I don't really see the need for `needle`; could do `const name = .....` directly.

Reply:
Size is used below to calculate the jump needed over the environment variable name.